File: app/lib/features/authentication/domain/entities/liveness_session_entity.dart
import 'package:dart_mappable/dart_mappable.dart';

import '../../../../core/enums/authentication/liveness_session_status.dart';

part 'liveness_session_entity.mapper.dart';

/// Sessão de prova de vida (detalhes brutos podem ficar no provedor / storage privado).
@MappableClass()
class LivenessSessionEntity with LivenessSessionEntityMappable {
  LivenessSessionEntity({
    required this.id,
    required this.userId,
    required this.provider,
    this.providerSessionId,
    required this.status,
    this.resultSummary,
    this.rawPayloadRef,
    required this.createdAt,
    this.completedAt,
  });

  final String id;
  final String userId;
  final String provider;
  final String? providerSessionId;
  final LivenessSessionStatus status;
  final Map<String, dynamic>? resultSummary;
  final String? rawPayloadRef;
  final DateTime createdAt;
  final DateTime? completedAt;
}

File: backend/functions/src/features/authentication/domain/entities/liveness_session_entity.ts
import {z} from "zod";

import {livenessSessionStatusSchema} from "./liveness_session_status.js";

/** Sessão de prova de vida (detalhes brutos no provedor ou storage privado). */
export const livenessSessionEntitySchema = z.object({
  id: z.string(),
  userId: z.string(),
  provider: z.string(),
  providerSessionId: z.string().optional(),
  status: livenessSessionStatusSchema,
  resultSummary: z.record(z.string(), z.unknown()).optional().nullable(),
  rawPayloadRef: z.string().optional().nullable(),
  createdAt: z.coerce.date(),
  completedAt: z.coerce.date().optional().nullable(),
});

export type LivenessSessionEntity = z.infer<typeof livenessSessionEntitySchema>;
